scripts/pylib/pytest-device-adapter/.pre-commit-config.yaml

```diff
@@ -0,0 +1,36 @@
+repos:
```

Comment:
> how is this file going to be used exactly?

If you are interested in how `pre-commit` works, you can find more information here: https://pre-commit.com/. We used it a lot during our work and it was very helpful. In general we decided to use more linters (like `flake8`, `isort` or `mypy`), because that helped us to maintain our code cleaner. In our initial idea we thought that it could be valuable to use them also in the further development and in CI pipelines - that's why we placed in this PR `.pre-commit-config.yaml` and `tox.ini` files. We don't have enough resources now to change and apply coding guidelines in the whole Twister project. So, if it is unacceptable to you to have different one for this pytest plugin, then I'll remove all those extra files from this PR.